### DISCUSSION TITLE
Allow disabled payment gateway to be conditionally enabled

### DISCUSSION THREAD

**Discussion by paulg-rymera** [woocommerce/woocommerce#50113] [Checkout Flow]
I'd just like to raise this question as it seems it's currently not possible in checkout block.

We have a feature in our plugin that allows store owners to map a role to payment gateways which would then be available for users of that role on checkout even if the payment gateway is _**disabled**_. This works in the legacy checkout form through available hooks but not in checkout block.

The implementation in the current checkout block allows for filtering _currently enabled_ payment gateways to be hidden but what we need is the opposite so, checking into how payment gateways are loaded in the checkout block, we found that there's no available hooks to programmatically _enable_ a disabled payment gateway.

So, the question is, would it be possible to implement a way to enable a _disabled_ payment gateway programmatically? Like maybe introduce a hook on `woocommerce/includes/abstracts/abstract-wc-payment-gateway.php:219` or something.

Thanks!

**Comment by senadir** [comment]
Hey! can you share a code example of how you're doing this using shortcode Checkout?

In Checkout block you can modify what payment methods might be available via `WC()->payment_gateways->get_available_payment_gateways()`

**Reply by paulg-rymera** [reply]
Doesn't work because the payment gateway needs to be enabled in the database. What we're doing in legacy checkout is something like:

```
//$user_payment_gateways = $this->get_wholesale_user_payment_gateways( $user_id, $user_wholesale_role[0] );
$user_payment_gateways = array( 'bacs', 'cod' );

if ( empty( $user_payment_gateways ) ) {
    return $available_gateways;
}

$all_payment_gateways = WC()->payment_gateways()->payment_gateways;
$filtered_gateways    = array();

foreach ( $all_payment_gateways as $gateway ) {
    if ( in_array( $gateway->id, $user_payment_gateways ) ) { // phpcs:ignore
        /**
         * Filtered gateways.
         *
         * @var WC_Payment_Gateway $gateway
         */
        $gateway->enabled = 'yes';

        $filtered_gateways[ $gateway->id ] = $gateway;
    }
}

if ( ! empty( $filtered_gateways ) ) {
    return $filtered_gateways;
}
```

This doesn't work in checkout block though.

**Reply by jkohlbach** [reply]
@senadir just checking in... did this get addressed/raised? It's definitely a regression. It's also blocking our 100% integration between Wholesale Suite and Checkout Block and is one of the last things on our list.

**Comment by jkohlbach** [comment]
@jamesckemp has this been taken care of in core yet?
